### PR TITLE
[CLI-1507] more accurate error message for users who has no permissions to describe a topic

### PR DESCRIPTION
**Pull request by MuweiHe**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

This is an internal change.
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

This message used to say `api-key doesn't have the acl` which is not accurate, as it's the user who has the acl, and api-keys only grants users such permissions. Besides acls, role-bindings could also be a factor that user has no permission to a topic. Updating the error message. 

References
----------

https://confluentinc.atlassian.net/browse/CLI-1507

Test & Review
-------------

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

**Review by brianstrauch — CHANGES_REQUESTED** [review]
Can you write some tests?

**Review by sgagniere — APPROVED** [review]
+1

**Comment by MuweiHe** [comment]
> Can you write some tests?

There's no mock backend for topic produce and consume thus we have no tests for them.

File: internal/pkg/errors/error_message.go [hunk 1]
```diff
@@ -130,36 +130,36 @@ const (
...
+	EncryptionKeySupportErrorMsg                     = "BYOK via `--encryption-key` is only available for GCP. Use `confluent byok create` to register AWS and Azure keys."
+	CKUMoreThanZeroErrorMsg                          = "`--cku` value must be greater than 0"
+	CKUMoreThanOneErrorMsg                           = "`--cku` value must be greater than 1 for High Durability"
+	ClusterResizeNotSupportedErrorMsg                = "failed to update kafka cluster: cluster resize is only supported on dedicated clusters"
+	CloudRegionNotAvailableErrorMsg                  = `"%s" is not an available region for "%s"`
+	CloudRegionNotAvailableSuggestions               = "To view a list of available regions for \"%s\", use `confluent kafka region list --cloud %s`."
+	CloudProviderNotAvailableErrorMsg                = `"%s" is not an available cloud provider`
+	CloudProviderNotAvailableSuggestions             = "To view a list of available cloud providers and regions, use `confluent kafka region list`."
+	TopicDoesNotExistOrMissingPermissionsErrorMsg    = `topic "%s" does not exist or you do not have the ACLs/role bindings required to describe it`
```
Comment (brianstrauch) [inline internal/pkg/errors/error_message.go:148]:
```suggestion
	TopicDoesNotExistOrMissingPermissionsErrorMsg    = `topic "%s" does not exist or you do not have the ACLs or role bindings required to describe it`
```

File: internal/pkg/errors/error_message.go [hunk 2]
```diff
@@ -130,36 +130,36 @@ const (
...
+	TopicDoesNotExistOrMissingPermissionsSuggestions = "To list topics for the cluster \"%s\", use `confluent kafka topic list --cluster %s`.\nTo list ACLs use `confluent kafka acl list --cluster %s`.\nTo list role bindings use `confluent rbac role-binding list`."
```
Comment (brianstrauch) [inline internal/pkg/errors/error_message.go:149]:
```suggestion
	TopicDoesNotExistOrMissingPermissionsSuggestions = "To list topics for the cluster \"%s\", use `confluent kafka topic list --cluster %s`.\nTo list ACLs use `confluent kafka acl list --cluster %s`.\nTo list role bindings use `confluent iam rbac role-binding list`."
```

File: internal/cmd/kafka/command_topic_onprem.go [hunk 3]
```diff
@@ -43,7 +43,7 @@ func (c *authenticatedTopicCommand) validateTopic(adminClient *ckafka.AdminClien
 	}
 	if !foundTopic {
 		log.CliLogger.Tracef("validateTopic failed due to topic not being found in the client's topic list")
-		return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.TopicDoesNotExistOrMissingACLsErrorMsg, topic), fmt.Sprintf(errors.TopicDoesNotExistOrMissingACLsSuggestions, "<cluster-Id>", "<cluster-Id>", "<cluster-Id>"))
+		return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.TopicDoesNotExistOrMissingPermissionsErrorMsg, topic), fmt.Sprintf(errors.TopicDoesNotExistOrMissingPermissionsSuggestions, "<cluster-Id>", "<cluster-Id>", "<cluster-Id>"))
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_onprem.go:46]:
```suggestion
		return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.TopicDoesNotExistOrMissingPermissionsErrorMsg, topic), fmt.Sprintf(errors.TopicDoesNotExistOrMissingPermissionsSuggestions, "<cluster-id>", "<cluster-id>", "<cluster-id>"))
```

File: internal/pkg/errors/error_message.go [hunk 4]
```diff
@@ -130,36 +130,36 @@ const (
...
+	EncryptionKeySupportErrorMsg                     = "BYOK via `--encryption-key` is only available for GCP. Use `confluent byok create` to register AWS and Azure keys."
+	CKUMoreThanZeroErrorMsg                          = "`--cku` value must be greater than 0"
+	CKUMoreThanOneErrorMsg                           = "`--cku` value must be greater than 1 for High Durability"
+	ClusterResizeNotSupportedErrorMsg                = "failed to update kafka cluster: cluster resize is only supported on dedicated clusters"
+	CloudRegionNotAvailableErrorMsg                  = `"%s" is not an available region for "%s"`
+	CloudRegionNotAvailableSuggestions               = "To view a list of available regions for \"%s\", use `confluent kafka region list --cloud %s`."
+	CloudProviderNotAvailableErrorMsg                = `"%s" is not an available cloud provider`
+	CloudProviderNotAvailableSuggestions             = "To view a list of available cloud providers and regions, use `confluent kafka region list`."
+	TopicDoesNotExistOrMissingPermissionsErrorMsg    = `topic "%s" does not exist or you do not have the ACLs or role bindings required to describe it`
```
Comment (brianstrauch) [inline internal/pkg/errors/error_message.go:148]:
```suggestion
	TopicDoesNotExistOrMissingPermissionsErrorMsg    = `topic "%s" does not exist or user does not have the ACLs or role bindings required to describe it`
```

File: internal/pkg/errors/error_message.go [hunk 5]
```diff
@@ -130,36 +130,36 @@ const (
...
+	TopicDoesNotExistOrMissingPermissionsSuggestions = "To list topics for the cluster \"%s\", use `confluent kafka topic list --cluster %s`.\nTo list ACLs use `confluent kafka acl list --cluster %s`.\nTo list role bindings use `confluent iam rbac role-binding list`."
```
Comment (brianstrauch) [inline internal/pkg/errors/error_message.go:149]:
```suggestion
	TopicDoesNotExistOrMissingPermissionsSuggestions = "To list topics for Kafka cluster \"%s\", use `confluent kafka topic list --cluster %s`.\nTo list ACLs use `confluent kafka acl list --cluster %s`.\nTo list role bindings use `confluent iam rbac role-binding list`."
```

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**